src/fragment-generation-utils.js

```diff
@@ -602,13 +664,15 @@ const reverseString = (string) => {
 /**
  * Determines whether the conditions for an exact match are met.
  * @param {Range} range - the range for which a fragment is being generated.
- * @return {boolean} - true if exact matching (i.e., only textStart) can be
- *     used; false if range matching (i.e., both textStart and textEnd) must be
- *     used.
+ * @return {boolean | TIMEOUT_SENTINEL} - true if exact matching (i.e., only
+ *     textStart) can be used; false if range matching (i.e., both textStart and
+ *     textEnd) must be used, or TIMEOUT_SENTINEL if execution must halt.
```

Comment:
Honestly it might just be easier to throw a timeout error. We don't have anything relevant to return anyways, and we'll be certain that JavaScript execution will be halted.

Reply:
This was a great idea, thanks.